opt_einsum/paths.py

```diff
@@ -971,14 +970,112 @@ def _dp_compare_size(
 
     s = s1 | s2
     i = _dp_calc_legs(g, all_tensors, s, inputs, i1_cut_i2_wo_output, i1_union_i2)
-    mem = helpers.compute_size_by_dict(i, size_dict)
+    mem = compute_size_by_dict(i, size_dict)
     cost = max(cost1, cost2, mem)
     if cost <= cost_cap:
         if s not in xn or cost < xn[s][1]:
             if memory_limit is None or mem <= memory_limit:
                 xn[s] = (i, cost, (contract1, contract2))
 
 
+def _dp_compare_write(
```

Comment:
These functions have surpassed what is reasonable to expect for straight args. Can we either pass as kwargs, a named tuple, or a data class?

Please type them as well.

Reply:
In general I'd agree, but I do think in this one case, i.e. a tight loop where performance is very important (this exact place is the bottleneck for many codes), using plain args is the right call. E.g. even just calling the function with kwargs slows down the dp optimizer a few percent - more 'readable' options like named tuples would incur a higher overhead. In fact given all the variable names already match the function signature I'm personally not sure it would be clearer either?

I had missed that all these functions had been type hinted when rebasing though - will add those.